contrib/node/src/python/pants/contrib/node/tasks/node_resolve.py

```diff
@@ -160,4 +160,7 @@ def execute(self):
                                                   resolve_locally=True,
                                                   install_optional=True,
                                                   frozen_lockfile=False)
-          node_paths_local.resolved(target, results_dir)
+          if target.is_synthetic:
```

Comment:
It would be good to put a comment why we are checking if the target is synthetic and why were are resolving the path differently here.